tests/trainer/test_trainer.py

```diff
@@ -1179,3 +1180,22 @@ def test_row_log_interval(log_metrics_mock, tmpdir, train_batches, max_steps, lo
     trainer.fit(model)
     expected_calls = [call(metrics=ANY, step=s) for s in range(log_interval - 1, max_steps, log_interval)]
     log_metrics_mock.assert_has_calls(expected_calls)
+
+    
+@pytest.mark.parametrize(['input_arg', 'expected'], [
+    (None, PassThroughProfiler),
+    (SimpleProfiler(), SimpleProfiler),
+    (AdvancedProfiler(), AdvancedProfiler),
+    ('simple', SimpleProfiler),
+    ('advanced', AdvancedProfiler),
+])
+def test_trainer_profiler_correct_args(input_arg, expected):
+    kwargs = {'profiler': input_arg} if input_arg is not None else {}
+    trainer = Trainer(**kwargs)
+    assert isinstance(trainer.profiler, expected)
+
+
+def test_trainer_profiler_incorrect_str_arg():
+    with pytest.raises(ValueError, match=r'when passing string value for the `profiler` parameter of '
+                       '`Trainer`, it can only be `simple` or `advanced`'):
```

Comment:
```suggestion
    with pytest.raises(ValueError, match=r"When passing string value for the `profiler` parameter of"
                       "` Trainer`, it can only be 'simple' or 'advanced'"):
```